src/services/earn/earn-service.ts

```diff
@@ -978,10 +953,100 @@ export class EarnService implements IEarnService {
       ],
     });
 
-    return swapData;
+    const multicalls = [...previousCalls, ...calls.map(({ tx }) => tx), runSwapData];
+    const multicall = encodeFunctionData({
+      abi: companionAbi,
+      functionName: 'multicall',
+      args: [multicalls],
+    });
+
+    return multicall;
   }
+  private async simulateSwaps({
+    request,
+    txValidFor,
+    quotes,
+    previousCalls,
+  }: {
+    request: Pick<QuoteRequest, 'chainId' | 'sellToken' | 'buyToken' | 'order'>;
+    txValidFor: TimeString;
+    quotes: QuoteResponseWithTx<Record<string, any>>[];
+    previousCalls: Hex[];
+  }) {
+    const swapsTx: Hex[] = [];
+    for (const quote of quotes) {
+      // Swap adapter uses the zero address as the native token
+      const tokenInDistribution = isSameAddress(quote.sellToken.address, Addresses.NATIVE_TOKEN)
+        ? Addresses.ZERO_ADDRESS
+        : quote.sellToken.address;
+      const tokenOutDistribution = isSameAddress(quote.buyToken.address, Addresses.NATIVE_TOKEN)
+        ? Addresses.ZERO_ADDRESS
+        : quote.buyToken.address;
+
+      const sellOrderSwapData = encodeFunctionData({
+        abi: permit2AdapterAbi,
+        functionName: 'sellOrderSwap',
+        args: [
+          {
+            deadline: BigInt(calculateDeadline(txValidFor) ?? calculateDeadline('1w')),
+            tokenIn: tokenInDistribution as ViemAddress,
+            amountIn: BigInt(quote.sellAmount.amount),
+            nonce: 0n,
+            signature: '0x',
+            allowanceTarget: quote.source.allowanceTarget as ViemAddress,
+            swapper: quote.tx.to as ViemAddress,
+            swapData: quote.tx.data as Hex,
+            tokenOut: tokenOutDistribution as ViemAddress,
+            minAmountOut: BigInt(quote.minBuyAmount.amount),
+            transferOut: [{ recipient: EARN_VAULT_COMPANION.address(request.chainId) as ViemAddress, shareBps: 0n }],
```

Comment:
Actually, instead of always sending the tokens to the Companion, we should take a recipient. During withdrawals (and delayed withdrawals), where there can be a recipient, we could send the tokens directly to it